packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -224,6 +242,18 @@ export class Pipeline extends cdk.Construct implements cpapi.IPipeline {
     this.artifactBucket.grantReadWrite(identity);
   }
 
+  /**
+   * Returns all of the {@link CrossRegionScaffoldStack}s that were generated automatically
+   * when dealing with Actions that live in a different region than the Pipeline itself.
+   */
+  public get crossRegionScaffoldStacks(): { [region: string]: CrossRegionScaffoldStack } {
```

Comment:
Is this just in order to return a copy of the map? Not sure there's much value in it, especially given the value itself is a reference and is technically mutable...

Why do we need this in the public API at all? Who calls this?

If this is just for test, just cast to `any` and read the private

Reply:
Yes, this is to defensively copy the map.

I think we should make it available to customers - for example, if using Romain's `app-delivery`, you would want to put these in your CDK-deploying CodePipeline.

Reply:
Well, that's an interesting discussion. Remember that you have to explicitly set a region on the CFN Action, and then not provide it in the `crossRegionReplicationBuckets` property, for this Stack to be created. So in some way, you can trust this Stack to be there, and you know exactly the key(s) it will contain.